frontend/src/metabase/visualizations/lib/settings/graph.ts

```diff
@@ -207,16 +225,18 @@ export const GRAPH_DATA_SETTINGS = {
         .map(getOptionFromColumn);
 
       const addedMetrics = vizSettings["graph.metrics"];
-      const hasBreakout = vizSettings["graph.dimensions"].length > 1;
-      const addedMetricsCount = addedMetrics.length;
+      const hasBreakout =
+        vizSettings["graph.dimensions"] &&
+        vizSettings["graph.dimensions"].length > 1;
+      const addedMetricsCount = addedMetrics?.length ?? 0;
       const maxMetricsSupportedCount = getMaxMetricsSupported(card.display);
 
       const hasMetricsToAdd = options.length > addedMetricsCount;
       const canAddAnother =
         addedMetricsCount < maxMetricsSupportedCount &&
         hasMetricsToAdd &&
         !hasBreakout &&
-        addedMetrics.every(metric => metric != null);
+        addedMetrics?.every(metric => metric != null);
```

Comment:
In the previous version, it looks like `addedMetrics` can never be null since we can always call `addedMetrices.length`.

That makes the new version look wrong because we assume that it could be null.